pkg/operator/starter.go

```diff
@@ -50,62 +51,91 @@ const (
+		guestKubeConfig, err = client.GetKubeConfigOrInClusterConfig(guestKubeConfigString, nil)
+		if err != nil {
+			return err
+		}
+		guestKubeClient = kubeclient.NewForConfigOrDie(rest.AddUserAgent(guestKubeConfig, operatorName))
+
+		// Create all events in the GUEST cluster.
+		// Use name of the operator Deployment in the management cluster + namespace
+		// in the guest cluster as the closest approximation of the real involvedObject.
+		controllerRef, err := events.GetControllerReferenceForCurrentPod(ctx, controlPlaneKubeClient, controlPlaneNamespace, nil)
+		controllerRef.Namespace = guestNamespace
+		if err != nil {
+			klog.Warningf("unable to get owner reference (falling back to namespace): %v", err)
+		}
+		eventRecorder = events.NewKubeRecorder(guestKubeClient.CoreV1().Events(guestNamespace), operandName, controllerRef)
+	}
+
+	guestDynamicClient, err := dynamic.NewForConfig(guestKubeConfig)
 	if err != nil {
 		return err
 	}
 
-	csiControllerSet := csicontrollerset.NewCSIControllerSet(
-		operatorClient,
-		controllerConfig.EventRecorder,
+	// Client informers for the GUEST cluster.
+	guestKubeInformersForNamespaces := v1helpers.NewKubeInformersForNamespaces(guestKubeClient, guestNamespace, "")
+	guestConfigMapInformer := guestKubeInformersForNamespaces.InformersFor(guestNamespace).Core().V1().ConfigMaps()
+	guestNodeInformer := guestKubeInformersForNamespaces.InformersFor("").Core().V1().Nodes()
+
+	guestConfigClient := configclient.NewForConfigOrDie(rest.AddUserAgent(guestKubeConfig, operatorName))
+	guestConfigInformers := configinformers.NewSharedInformerFactory(guestConfigClient, 20*time.Minute)
+	guestInfraInformer := guestConfigInformers.Config().V1().Infrastructures()
+
+	// Create client and informers for our ClusterCSIDriver CR.
+	gvr := opv1.SchemeGroupVersion.WithResource("clustercsidrivers")
+	guestOperatorClient, guestDynamicInformers, err := goc.NewClusterScopedOperatorClientWithConfigName(guestKubeConfig, gvr, string(opv1.AWSEBSCSIDriver))
+	if err != nil {
+		return err
+	}
+
+	// Start controllers that manage resources in the MANAGEMENT cluster.
+	controlPlaneCSIControllerSet := csicontrollerset.NewCSIControllerSet(
+		guestOperatorClient,
+		eventRecorder,
 	).WithLogLevelController().WithManagementStateController(
 		operandName,
 		false,
 	).WithStaticResourcesController(
 		"AWSEBSDriverStaticResourcesController",
-		kubeClient,
-		dynamicClient,
-		kubeInformersForNamespaces,
-		assets.ReadFile,
+		controlPlaneKubeClient,
+		controlPlaneDynamicClient,
+		controlPlaneKubeInformersForNamespaces,
+		assetWithNamespaceFunc(controlPlaneNamespace),
 		[]string{
-			"storageclass_gp2.yaml",
-			"volumesnapshotclass.yaml",
-			"csidriver.yaml",
 			"controller_sa.yaml",
 			"controller_pdb.yaml",
-			"node_sa.yaml",
 			"service.yaml",
 			"cabundle_cm.yaml",
 			"rbac/attacher_role.yaml",
 			"rbac/attacher_binding.yaml",
 			"rbac/privileged_role.yaml",
 			"rbac/controller_privileged_binding.yaml",
```

Comment:
This reminds me: the CSI driver controller Deployment should not run as privileged and read from metadata service. The mgmt cluster can be in a different Region or not on AWS at all.

Therefore, we need to update the driver a bit + remove this RoleBinding also in standalone OCP, we can save us some node ports + port conflicts there too. I tried this with the driver Deployment yaml file:

- remove "hostNetwork: true":
- run the driver with `controller` as the first (!) argument.
- set env. var "AWS_REGION=us-east2". This value can be populated from (guest) Infrastructure object (i.e. a new hook).
- CSI driver on nodes can be then run with `node` as the first argument. The do need the metadata service.

Reply:
Thank you for the pointers. Done.